### DISCUSSION TITLE
Unit tests

### DISCUSSION THREAD

**Discussion by createdbymahmood** [sonnat/sonnat-ui#16] [General]
Hi, I was wondering if you want to write some unit tests and I'd like to help you with this.

**Comment by mimshins** [comment]
@createdbymahmood 
I was actually considering to add unit tests this year but delivering v2 kept me busy.
However, we can talk about it in this thread of discussion.

**Comment by mimshins** [comment]
Do you have any experiences with testing react components?
What set of tools have you used for this purpose?
I think we might have to benchmark other libraries (i.e Material-UI, Ant-Design, etc.) as well.

**Reply by createdbymahmood** [reply]
Yes I do.
I'm using [React Testing Library](https://github.com/testing-library/react-testing-library) and [jest](https://github.com/facebook/jest) to test React components, but I'm open to use other libraries as well.

**Reply by mimshins** [reply]
Awesome!
That'll do the work ^^

For the test cases itself, I believe it would be good practice to checkout the [Material-UI](https://github.com/mui/material-ui/tree/master/packages/mui-material/src)'s test cases, That'll lead us to the right direction.

For example: [Test Cases for Material-UI Button](https://github.com/mui/material-ui/tree/master/packages/mui-material/src)

**Reply by mimshins** [reply]
So, are you going to take responsibility for this task?
We can communicate further through Discord.

**Reply by createdbymahmood** [reply]
I do agree, MUI test cases are perfect.
Yes, I'm going to do that and will contact you soon.

**Reply by mimshins** [reply]
Awesome!
I'll create [an issue](https://github.com/sonnat/sonnat-ui/issues/17) from this discussion.

(BTW, here's my Discord ID: `mimshins#7777`)
